internal/auth/jwt.go

```diff
@@ -0,0 +1,52 @@
+package auth
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// jwtClaims represents the expected claims contained in the JWT token.
+type jwtClaims struct {
+	AccessID           string         `json:"access_id"`
+	Actor              map[string]any `json:"act,omitempty"`
+	AuthMethods        []string       `json:"amr,omitempty"`
+	AuthenticationTime int64          `json:"auth_time,omitempty"`
+	ClientID           string         `json:"cid,omitempty"`
+	ExpiresAt          int64          `json:"exp,omitempty"`
+	GrantType          string         `json:"grant,omitempty"`
+	IssuedAt           int64          `json:"iat,omitempty"`
+	Issuer             string         `json:"iss,omitempty"`
+	JWTID              string         `json:"jti,omitempty"`
+	NotBefore          int64          `json:"nbf,omitempty"`
+	Namespace          string         `json:"ns,omitempty"`
+	Scopes             []string       `json:"scp,omitempty"`
+	Subject            string         `json:"sub,omitempty"`
+}
+
+// unsafeParseJWT parses a JWT without verifying its signature and returns its claims.
+// WARNING: This is intentionally unsafe and must not be used for trust decisions.
+func unsafeParseJWT(token string) (*jwtClaims, error) {
```

Comment:
since we only need ExpiresAt, we could make that a private (unnamed) struct, and return only that, e.g.


```go
func unsafeGetJWTExpiry(token string) (time.Time, error) {
	var claims struct {
		ExpiresAt *jwt.NumericDate `json:"exp,omitempty"`
	}
	// ...
```

and that NumericDate comes from `github.com/go-jose/go-jose/v4/jwt"` - it is useful for parsing these UNIX timestamps - https://github.com/go-jose/go-jose/blob/9860c65054c4821d1e7c22200422b04181f58ebc/jwt/claims.go#L38-L42

Reply:
Updated the code to be like the suggestion, except the `jwt.NumericDate` which is not enough IMO to justify adding another library. Added the Unix -> time.Time conversion in the code.